lib/collector.js

```diff
@@ -88,6 +106,13 @@ class Collector {
   }
 }
 
+function getSettled(err = null, value = null) {
```

Comment:
It shouldn't be a separate function, it should be inside a class `Collector`.  And what must do this function? Function `set` takes the argument `value` not as an object.

Reply:
I made getSettled as static method in class  Collector before,  but @tshemsedinov said said that the function is better.
Function should return an object like Promise.allSettled() depending on whether the result is "fulfilled" or "rejected".